portfolio/src/main/java/com/google/sps/data/Comment.java

```diff
@@ -17,50 +17,111 @@
 import java.util.Date;
 import java.text.DateFormat;
 import java.lang.Comparable;
+import java.util.List;
+import java.util.ArrayList;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Key;
 import javax.servlet.http.HttpServletRequest;
 
 /** Represents a single comment from a user. */
 public class Comment implements Comparable<Comment> {
-  public static final String COMMENT_NAME = "name";
-  public static final String COMMENT_TEXT = "comment";
-  public static final String COMMENT_TIMESTAMP = "timestamp";
+  /** Key for the name of the commenter */
+  private static final String COMMENT_NAME = "name";
+  /** Key for the comment contents */
+  private static final String COMMENT_TEXT = "comment";
+  /** Key for the timestamp */
+  private static final String COMMENT_TIMESTAMP = "timestamp";
+
+  private static int commentCount = 0;
 
   private final String text;
   private final String name;
   private final long timestamp;
+  private final int id;
+  /** The key associated with the entity that keeps this comment persist in storage. */
+  private final Key key;
 
+  /** Will automatically add to persistent storage */
   private Comment(String text, String name) {
     this(text, name, System.currentTimeMillis());
   }
 
-  /** Create a comment with a posted date of given milliseconds from the epoch */
+  /** Create a comment with a posted date of given milliseconds from the epoch. 
+   *  Automatically adds the new comment to persistent storage */
   private Comment(String text, String name, long timestamp) {
     this.text = text;
     this.name = name;
     this.timestamp = timestamp;
+    this.key = persistStore();
+
+    // Dynamically assign ID
+    id = commentCount++;
+  }
+
+  /** Will NOT automatically add the comment to persistent storage */
+  private Comment(String text, String name, long timestamp, Key key) {
+    this.text = text;
+    this.name = name;
+    this.timestamp = timestamp;
+    this.key = key;
+
+    // Dynamically assign ID
+    id = commentCount++;
+  }
+
+  /** Adds the given comment to persistent storage */
+  private Key persistStore() {
```

Comment:
key topic: 
Single responsibility principle

Reply:
Ok -- I created `CommentPersistHelper` to serve as a wrapper for `Comment` persistence.